### PR TITLE
Add openspecfun

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/conda-forge/staged-recipes/issues/3973

Add openspecfun as a conda-forge package on macOS and Linux. This is to be used as a requirement of `julia`.

### CODE REVIEW

File: recipes/openspecfun/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  home: https://github.com/JuliaLang/openspecfun
+  license: MIT, Public Domain
+  license_file: LICENSE.md
+  summary: A collection of special mathematical functions
+  dev_url: https://github.com/JuliaLang/openspecfun
+
+extra:
+  recipe-maintainers:
+    - dfornika
+    - jakirkham
```
Comment (jakirkham) [inline recipes/openspecfun/meta.yaml]: Added you here @dfornika. Please let me know if that is ok.

File: recipes/openspecfun/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - toolchain
+    - gcc             # [osx]
+    - libgfortran     # [linux]
+    - openlibm 0.5.4
```
Comment (jakirkham) [inline recipes/openspecfun/meta.yaml]: Pinned tightly for now to match what Julia is doing in their own codebase. Also we are unclear about the ABI compatibility of this `openlibm` ATM. Can revisit relaxing this pinning once we know more.

File: recipes/openspecfun/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  run:
+    - libgfortran
+    - openlibm 0.5.4
+
+test:
+  commands:
+    - test -f "${PREFIX}/include/Faddeeva.h"                             # [unix]
```
Comment (isuruf) [inline recipes/openspecfun/meta.yaml]: I'm not sure whether this is the best place to have this as we might package `Faddeeva` separately.

Comment (jakirkham) [reply recipes/openspecfun/meta.yaml]: If we do, we can always make this depend on a `faddeeva` package.

File: recipes/openspecfun/meta.yaml (hunk 1 above)
Comment (dfornika) [reply recipes/openspecfun/meta.yaml]: Yes, that's ok.

**Pull request merged**